corehq/apps/case_search/temp/fix_child_cases.py

```diff
@@ -0,0 +1,115 @@
+def update_script(case_ids_to_exclude=[]):
+    from corehq.form_processor.models import CommCareCase
+    from corehq.apps.locations.models import SQLLocation
+    from casexml.apps.case.mock import CaseBlock
+    from corehq.apps.hqcase.utils import submit_case_blocks
+    import os
+
+    def submit_cases(case_blocks):
+        submit_case_blocks(
+            [cb.as_text() for cb in case_blocks],
+            domain=DOMAIN,
+            device_id='system',
+        )
+
+    DOMAIN = 'alafiacomm'
+    CHILD_CASE_TYPE = 'membre'
+    BULK_SIZE = 50
+
+    total_case_ids = CommCareCase.objects.get_case_ids_in_domain(domain=DOMAIN, type=CHILD_CASE_TYPE)
+    case_ids = [case_id for case_id in total_case_ids if case_id not in case_ids_to_exclude]
+
+    total_cases = len(case_ids)
+    cases_checked = -1
+
+    error_log_file_path = os.path.expanduser("~/chris_script_error.log")
+    success_log_file_path = os.path.expanduser("~/chris_script_success.log")
+    with open(success_log_file_path, "a") as success_logfile:
+        with open(error_log_file_path, "a") as logfile:
+
+            case_blocks = []
+            for child_case in CommCareCase.objects.iter_cases(case_ids, domain=DOMAIN):
+                # --- Statistics
+                cases_checked += 1
+                progress = round((cases_checked / total_cases) * 100, 3)
+                print(f"{progress}% complete")
+                # --------------
+
+                """
+                    Step 2 - Make sure that the case properties of both the child and parent cases correspond to te
+                    village's hierarchy. The owner_id of both cases is the village id
+                """
+                try:
+                    parent_case = child_case.parent
+                    village_id = parent_case.owner_id
+                except Exception:
+                    logfile.write(f"Skipped {child_case.case_id}. Reason: No parent case\n")
+                    continue
+
+                try:
+                    village = SQLLocation.objects.filter(location_id=village_id, domain=DOMAIN).first()
```

Comment:
```suggestion
                    village = SQLLocation.objects.get(location_id=village_id, domain=DOMAIN)
```
Might as well get the error as soon as possible, instead of setting village to `None` if it doesn't exist.

Reply:
Good idea